### ISSUE TITLE
Speed up bytes creation from list and tuple

### ISSUE THREAD

**Issue by blhsing** [python/cpython#128213]
# Feature or enhancement

### Proposal:

The performance of bytes creation from list and tuple can benefit from the same fast path using `PySequence_Fast_ITEMS` done for bytearray in gh-91149.

I have a PR ready.

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

### Linked PRs
* gh-128214
* gh-132590

**Comment by serhiy-storchaka** [comment]
How much does it benefit? Please show some microbenchmark results.

**Comment by blhsing** [comment]
> How much does it benefit? Please show some microbenchmark results.

I just finished updating my benchmark results in the PR, but I'll cross post them here:

Benchmark using `python -m timeit -n 100 -s 'a = __import__("random").choices(range(256), k=1000000)' 'bytes(a)'` showing a ~31% reduction in time consumed:
```
100 loops, best of 5: 3.21 msec per loop # current
100 loops, best of 5: 2.23 msec per loop # this PR
```
With `k=10000`, a ~30% time reduction:
```
100 loops, best of 5: 31.7 usec per loop # current
100 loops, best of 5: 22.3 usec per loop # this PR
```
With `k=100`, a ~27% time reduction:
```
100 loops, best of 5: 410 nsec per loop # current
100 loops, best of 5: 299 nsec per loop # this PR
```

**Comment by sergey-miryanov** [comment]
@blhsing @eendebakpt Is there anything else we still need to do?

**Comment by eendebakpt** [comment]
> [@blhsing](https://github.com/blhsing) [@eendebakpt](https://github.com/eendebakpt) Is there anything else we still need to do?


Yes. There is a free-threading bug and potentially a performance gain. The PR https://github.com/python/cpython/pull/132590 is ready for review.
